src/layouts/components/searchbar/SearchBar.tsx

```diff
@@ -0,0 +1,148 @@
+import { useMemo, useState } from "react";
+import { debounce } from "lodash";
+import { shift, autoUpdate, size, useFloating } from "@floating-ui/react-dom";
+import { Combobox, Transition } from "@headlessui/react";
+import clsx from "clsx";
+import { FiSearch } from "react-icons/fi";
+import { IoCloseCircleSharp } from "react-icons/io5";
+import { SearchResult, SearchResultTable } from "./SearchResult";
+
+export function SearchBar(): JSX.Element {
+  const [isSearching, setIsSearching] = useState<boolean>(false);
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+  const [searchString, setSearchString] = useState();
+  const [searchResults, setSearchResults] = useState<
+    SearchResult[] | undefined
+  >(undefined);
+
+  const [selected, setSelected] = useState<SearchResult>();
+  const { x, y, reference, floating, strategy, refs } = useFloating({
+    placement: "bottom-end",
+    middleware: [
+      shift(),
+      size({
+        apply({ rects }) {
+          if (
+            refs.floating.current !== null &&
+            refs.floating.current !== undefined
+          ) {
+            Object.assign(refs.floating.current.style, {
+              minWidth: "325px",
+              width: `${rects.reference.width}px`,
+            });
+          }
+        },
+      }),
+    ],
+    whileElementsMounted: autoUpdate,
+  });
+
+  async function changeHandler(value): Promise<void> {
+    const query = value.trim();
+    setSearchString(query);
+    setSelected({ title: query, url: "", type: "Query" });
+    if (query.length > 0) {
+      setIsSearching(true);
+      const results = await getSearchResults(query);
+      setSearchResults(results);
+      setIsSearching(false);
+    } else {
+      setSearchResults([]);
+    }
+  }
+
+  const onChangeDebounceHandler = useMemo(
+    () => debounce((event) => changeHandler(event.target.value), 200),
+    []
+  );
+
+  const onSelect = (result?: SearchResult): void => {
+    setSelected(result);
+    // TODO add on select action
+  };
+
+  return (
+    <Combobox value={selected} onChange={onSelect} nullable>
+      <div className="flex w-full px-5 md:px-10 lg:px-[316px] items-center justify-self-center mx-auto my-10">
```

Comment:
let's not use horizontal padding to set the width of search bar when view port > `lg`, we can go with %. Currently it will go too wide when screen is also wide
<img width="2559" alt="image" src="https://user-images.githubusercontent.com/40191153/200111422-3ff99939-814c-48c9-b146-f68600c3a714.png">

Reply:
or add `container` that defines the max width